### ISSUE TITLE
Support for vscode

### ISSUE THREAD

**Issue by SethPoulsen** [brownplt/atom-language-pyret#2]
I've just ported this to work for Visual Studio Code and want to publish it. Would you guys like to have it published on the brownplt page, or should I just do it on my own page? 

Any preference of what I name it when I publish to the vscode extension repositories? 

Thanks!

**Comment by SethPoulsen** [comment]
Also, I would like to be able to use the Pyret logo when I publish the extension. Is that possible?

**Comment by SethPoulsen** [comment]
For now, I've uploaded to here: https://github.com/SethPoulsen/vscode-language-pyret.

**Comment by jpolitz** [comment]
@SethPoulsen this is awesome! I just started trying out VSCode myself, so this is cool timing for me :-)

I made a repo under brownplt, if you are interested in submitting pull requests there we will happily merge them – what you have looks great, I just downloaded it and tried it out.

https://github.com/brownplt/vscode-language-pyret

Just to double-check, where do you want to use the logo? Is it for icons and stuff in the published extension, in the README, etc?

**Comment by SethPoulsen** [comment]
@jpolitz Glad you like it!

Yea, it would be cool if the logo would show up with the extension on https://marketplace.visualstudio.com, and if it could show up next to the file names in the file explorer (side panel) in VSCode.

**Comment by SethPoulsen** [comment]
Also, I can't figure out how to get the new repository to allow me to open a pull request. It seems to be disallowing it since it is completely empty.

**Comment by jpolitz** [comment]
I added an initial commit to that repo – let me know if that works for you.

We're happy for you to use the logo for this, and thanks for asking first! There's a file called "pyret-icon.png" that we use for favicons that might be especially good for file logo since it avoids trying to scale down the details of the eyepatch:

https://github.com/brownplt/pyret-lang/blob/master/img/pyret-icon.png

Let us know if you have any other questions.

**Comment by SethPoulsen** [comment]
Its in the vscode extension marketplace!
![screen shot 2018-03-11 at 12 49 22 am](https://user-images.githubusercontent.com/25016959/37251115-808a2190-24c6-11e8-952f-961858d70747.png)

Should be easy for anyone to install now. Unfortunately, getting the icon to show next to the file name is more complicated, I have submitted an issue here to make it happen: https://github.com/vscode-icons/vscode-icons/issues/1502

**Comment by JimiC** [comment]
@jpolitz Any chance we can get our 'hooks' (hands) on an `SVG` version of the logo? We at `vscode-icons` prefer `SVG` as they are more digestible!!! 😆

**Comment by jpolitz** [comment]
Indeed! @JimiC there are svg versions that we use to generate the .pngs at https://github.com/brownplt/pyret-lang/tree/master/img/

**Comment by JimiC** [comment]
Ah I see. I admit I was too lazy to do a search. My apologies.

**Comment by shriram** [comment]
Awesome! Now I almost wish I used Visual Studio. (-:

What is the user experience with it, btw?

**Comment by JimiC** [comment]
Great and it's getting better. But I'm biased so I don't know if my opinion counts.

**Comment by SethPoulsen** [comment]
Visual Studio Code is awesome. Of course, I wouldn't have bothered making the Pyret extension if it wasn't.

**Comment by SethPoulsen** [comment]
Shoutout to @JimiC for making the icons happen. Looks awesome!
![screen shot 2018-03-21 at 5 30 06 pm](https://user-images.githubusercontent.com/25016959/37743007-b51f902e-2d2d-11e8-9cca-b8324cdb9422.png)

**Comment by jpolitz** [comment]
Cool! For folks seeing this who don't know the details, this is enabled by installing the vscode-icons package (you can just search for "vscode-icons" in the package installer within vscode). I'm new to vscode, but from my reading this is a pretty standard theming package, recommended by the vscode team:

https://code.visualstudio.com/blogs/2016/09/08/icon-themes

Thanks @JimiC and @SethPoulsen !
